annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+            objs_ids=file_ids, cache_used=mock_cache
+        )
+
+        mock_get_file_names_by_request.assert_called_once_with(
+            file_ids=list(not_cached_file_ids), tenant=tenant, token=token
+        )
+
+        mock_cache.update.assert_called_once_with(file_names_from_requests)
+
+        expected_result = {**cached_file_names, **file_names_from_requests}
+
+        assert result == expected_result
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
+    metric.task_from = datetime(2024, 1, 1)
+    metric.task_to = datetime(2024, 2, 1)
+    metric.agreement_metric = True
+    return metric
+
+
+@pytest.fixture
+def mock_stats(mock_task, mock_metric):
+    stat1 = MagicMock()
+    stat1.task = mock_task
+    stat1.task_id = 1
+    stat1.created = datetime(2024, 1, 1, 12, 0, 0)
+    stat1.updated = datetime(2024, 1, 2, 12, 0, 0)
+    stat1.task.user_id = 2
+    stat1.task.file_id = 3
+    stat1.task.pages = [1, 2, 3]
+    stat1.task.status.value = "completed"
+    stat1.task.agreement_metrics = [mock_metric]
+
+    stat2 = MagicMock()
+    stat2.task = mock_task
+    stat2.task_id = 2
+    stat2.created = datetime(2024, 1, 3, 12, 0, 0)
+    stat2.updated = datetime(2024, 1, 4, 12, 0, 0)
+    stat2.task.user_id = 2
+    stat2.task.file_id = 3
+    stat2.task.pages = [1, 2, 3]
+    stat2.task.status.value = "completed"
+    stat2.task.agreement_metrics = [mock_metric]
+
+    stat3 = MagicMock()
+    stat3.task = mock_task
+    stat3.task_id = 3
+    stat3.created = datetime(2024, 1, 5, 12, 0, 0)
+    stat3.updated = datetime(2024, 1, 6, 12, 0, 0)
+    stat3.task.user_id = 2
+    stat3.task.file_id = 3
+    stat3.task.pages = [1, 2, 3]
+    stat3.task.status.value = "completed"
+    stat3.task.agreement_metrics = [mock_metric]
+
+    return [stat1, stat2, stat3]
+
+
+@pytest.fixture
+def mock_db(mock_stats):
+    mock_db = MagicMock()
+    mock_query = MagicMock()
+    mock_query.filter.return_value.filter.return_value.all.return_value = (
+        mock_stats
+    )
+    mock_db.query.return_value = mock_query
+
+    return mock_db
+
+
+@pytest.fixture
+def mock_user_name():
+    return {2: "user2"}
+
+
+@pytest.fixture
+def mock_file_name():
+    return {3: "file3"}
+
+
+@pytest.mark.parametrize(
+    ("schema", "expected_filename"),
+    (
+        (
+            MagicMock(
+                user_ids=[2],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            "annotator_stats_export_20240816.csv",
+        ),
+        (
+            MagicMock(
+                user_ids=[999],
+                date_from=datetime(2024, 1, 1),
+                date_to=datetime(2024, 1, 6),
+            ),
+            "annotator_stats_export_20240816.csv",
+        ),
+    ),
+)
+def test_create_export_csv(
+    schema: Mock,
+    expected_filename: str,
+    mock_db: Mock,
+    mock_user_name: Mock,
+    mock_file_name: Mock,
```

Comment:
These Mocks are dictionaries, so it should be reflected in typing.
Since they're created for this test, they shouldn't be fixtures. They can be local variables.